app.js

```diff
@@ -0,0 +1,299 @@
+const {
+  bootstrap,
+  helpers,
+  docker,
+} = require("@kaholo/plugin-library");
+const { promisify } = require("util");
+const exec = promisify(require("child_process").exec);
+
+const HELM_CLI_NAME = "helm";
+const HELM_IMAGE_NAME = "alpine/helm";
+
+async function install(pluginParameters) {
+  const { kubeCertificate } = pluginParameters;
+
+  let result;
+
+  await helpers.temporaryFileSentinel(
+    [kubeCertificate],
+    async (certificateFilePath) => {
+      const [certificatePath, certificateFileName] = splitDirectory(certificateFilePath);
+      const certificateVolumeDefinition = docker.createVolumeDefinition(certificatePath);
+      const chartVolumeDefinition = docker.createVolumeDefinition(pluginParameters.chartDirectory);
+      const volumeDefinitions = [
+        certificateVolumeDefinition,
+        chartVolumeDefinition,
+      ];
+
+      const authenticationParameters = generateAuthenticationParameters({
+        certificateFilePath: `$${certificateVolumeDefinition.mountPoint.name}/${certificateFileName}`,
+        kubeToken: pluginParameters.kubeToken,
+        kubeApiServer: pluginParameters.kubeApiServer,
+        kubeUser: pluginParameters.kubeUser,
+      });
+
+      const releaseNameParameters = generateReleaseNameParameters({
+        releaseName: pluginParameters.releaseName,
+        generateName: pluginParameters.generateName,
+      });
+
+      const installationParameters = generateInstallationParameters({
+        namespace: pluginParameters.namespace,
+        valuesOverride: pluginParameters.valuesOverride,
+      });
+
+      const dockerEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.mountPoint.name]: cur.mountPoint.value,
+        }),
+        {},
+      );
+
+      const shellEnvironmentalVariables = volumeDefinitions.reduce(
+        (acc, cur) => ({
+          ...acc,
+          [cur.path.name]: cur.path.value,
+        }),
+        dockerEnvironmentalVariables,
+      );
+
+      const helmCommand = `\
+install \
+${releaseNameParameters.join(" ")} \
+$${chartVolumeDefinition.mountPoint.name} \
+${installationParameters.join(" ")} \
+${authenticationParameters.join(" ")}`;
+
+      const command = docker.buildDockerCommand({
+        command: helmCommand,
+        image: HELM_IMAGE_NAME,
+        environmentVariables: dockerEnvironmentalVariables,
+        volumeDefinitionsArray: volumeDefinitions,
+      });
+
+      result = await exec(command, {
+        env: shellEnvironmentalVariables,
+      });
+    },
+  );
+
+  const { stdout, stderr } = result;
+
+  if (!stdout && stderr) {
+    throw new Error(stderr);
+  }
+
+  return stdout;
+}
+
+async function uninstall(pluginParameters) {
+  const { chartName, kubeCertificate } = pluginParameters;
+
+  let result;
+
+  await helpers.temporaryFileSentinel(
+    [kubeCertificate],
+    async (certificateFilePath) => {
```

Comment:
We can extract this function too, it's very similar to the `temporaryFileSentinel` callback function in the `install` function, we can merge these two together instead of writing the same code twice.

Reply:
Fixed